### ISSUE TITLE
IMAP folder name in () select failed error

### ISSUE THREAD

**Issue by sadrainsky** [bagder/curl#517]
IMAP folder name in () select failed error

curl  "imap://localhost:143/(test)" -X "FETCH 1 BODY[TEXT]" -u user:password
curl: (67) Select failed

-> 
imap.c

before :  
  /* Send the SELECT command */
  result = imap_sendf(conn, "SELECT %s", mailbox);

after : 
  /* Send the SELECT command */
  result = imap_sendf(conn, "SELECT \"%s\"", mailbox);

**Comment by captain-caveman2k** [comment]
Hi,

Thanks for you bug report and suggested fix. Do you think we should only quote the selected item when it contains characters that need it?

Off the top of my head the unit test will also need updating if we always quote the item.

Steve

**Comment by bagder** [comment]
what does the protocol spec say about the quotes?

**Comment by captain-caveman2k** [comment]
In section 5.1 Mailbox naming it states:

*Any character which is one of the atom-specials (see the Formal Syntax) will require that the mailbox name be represented as a quoted string or literal.*

It also states that:

*The case-insensitive mailbox name INBOX is a special name reserved to mean "the primary mailbox for this user on this server".  The interpretation of all other names is implementation-dependent.*

As such I wouldn't like to "just" quote the mailbox name when it isn't required - as servers may or may not interpret that differently.

In my opinion it would be better to pass the mailbox name to a function that escapes it - this means we don't need to update the existing tests, won't potentially break odd server implementations and also allows use to support International Naming Convention at some point in the future as well ;-)

Elsewhere the RFC mentions that the mailbox format uses "astring" syntax which is an atom or a string.

The "Formal Syntax" then goes on to say that "atom-specials" are:

 "(", ")", "{", SPACE, CTL, "%", "*", "\" and "]" - hopefully I haven't missed any as I looked each "type" up ;-)

**Comment by captain-caveman2k** [comment]
I've looked at the code this evening and it appears that we already do something similar to what I suggested earlier in imap_atom().

However, we aren't looking for the full range of "atom-special" characters just \, " and SPACE. If the function finds a \ or " it escapes it and if it finds a space it encloses the whole string with quotes.

As such I think we should modify that function to also look for the other characters.

I'll try and take a look at the weekend but as folk may have noticed my email and/or autobuild VMs have gone down and I need to take a look at those :(
